ecommerce/migrations/0020_product_unique_object_id_validated.py

```diff
@@ -0,0 +1,21 @@
+# Generated by Django 3.2.14 on 2022-08-02 20:39
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("ecommerce", "0019_db_protect_product"),
+    ]
+
+    operations = [
+        migrations.AddConstraint(
+            model_name="product",
+            constraint=models.UniqueConstraint(
+                condition=models.Q(("is_active", True)),
+                fields=("object_id", "is_active"),
+                name="unique_object_id_validated",
+            ),
+        ),
```

Comment:
So if there's currently duplicate products (I know RC does/did have one at one point). We can deal with this either by deleting the conflicts prior to the migration or deal with it in the migration.

Reply:
Yeah, I figured it would be best to deal with them pre-migration.  What is likely to happen if they are left and I perform the migration?

Reply:
the migration will fail because existing data violates the constraint

Reply:
So would you suggest one method of resolving this over another?

Reply:
Production has no products, so if you're fine cleaning up RC before this gets deployed you can just go ahead with that.